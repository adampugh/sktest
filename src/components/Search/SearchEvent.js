import React from 'react';
import { Link } from 'react-router-dom';
import { FaCalendarAlt, FaMapMarkerAlt } from 'react-icons/fa';


const SearchEvent = ({ name, date, desc, img, venue, id }) => (
    <div className="searchEvent">
        <img src={img} alt={name} />
        <div className="searchEvent__text">
            <h2 className="title">{name}</h2>
            <p className="searchEvent__text__desc">{desc}</p>
            <div className="searchEvent__text__date">
                <FaCalendarAlt />
                <p>{date}</p>
            </div>
            <div className="searchEvent__text__venue">
                <FaMapMarkerAlt />
                <p>{venue}</p>
            </div>
            <Link to={`/event/${id}`}>
                <button className="btn">view details</button>
            </Link>
        </div>
    </div>
);

export default SearchEvent;