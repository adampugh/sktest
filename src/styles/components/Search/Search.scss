.search__hero {
    background: $sk-blue;
    width: 100%;
    padding: 20px 0;
}

.search__hero__logo {
    max-width: 500px;
    width: 100%;
    margin: 0 auto;
    display: block;
}

.search__hero__input {
    display: block;
    margin: 0 auto;
    border: none;
    border-radius: 5px;
    width: 80%;
    max-width: 500px;
    padding: 20px;
    text-align: center;
}

.search__events {
    background: $sk-grey;
    height: 100%;
    padding: 40px 0;
}

.search__events__grid {
    display: grid;
    grid-template-columns: 1fr;
    grid-gap: 20px;
}


@media (min-width: $desktop-breakpoint) {
    .search__events__grid {
        grid-template-columns: 1fr 1fr 1fr;
    }
}