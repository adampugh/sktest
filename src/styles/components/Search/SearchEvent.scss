.searchEvent {
    background: white;
    text-align: center;
    max-width: 100%;
    @include boxshadow;
    h2 {
        color: #121212;
        margin-bottom: 10px;
    }
    img {
        width: 100%;
        max-width: 100%;
    }
    
}

.searchEvent__text__date, .searchEvent__text__venue {
    text-align: left;
    display: block;
    margin: 10px 0;
    p {
        display: inline;
        margin-left: 10px;
    }
}

.searchEvent__text__desc {
    height: 100px;
}

.searchEvent__text {
    padding: 20px;
    button {
        margin-top: 10px;
        &:hover {
            @include boxshadow;
            cursor: pointer;
        }
    }
}