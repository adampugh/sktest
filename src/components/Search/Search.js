import React, { Component } from 'react';
import skiddle from '../../apis/skiddle';

import Logo from '../../assets/logo.png';
import SearchEvent from './SearchEvent';

class Search extends Component {
    state = {
        keyword: '',
        events: []
    }

    handleInput = (e) => {
        this.setState({
            keyword: e.target.value
        });
    }

    handleEnter = (e) => {
        if (e.key === 'Enter') {
            this.onSubmit();
        }
    }

    onSubmit = () => {
        const { keyword } = this.state;

        skiddle.get(`/events/search/?api_key=${process.env.REACT_APP_API_KEY}&keyword=${keyword}`)
            .then(response => {
                this.setState({
                    events: response.data.results
                });
            }).catch(error => {
                console.log(error);
            });
    }


    render() {
        const { keyword, events } = this.state;

        return (
            <div className="search">
                <div className="search__hero">
                    <div className="container">
                        <img className="search__hero__logo" src={Logo} alt="logo" />
                        <input 
                            className="search__hero__input" 
                            placeholder="Search for events"
                            value={keyword}  
                            onChange={this.handleInput}
                            onKeyPress={this.handleEnter}  
                        />
                    </div>
                </div>
                <div className="search__events">
                    <div className="search__events__grid container">
                    {
                        events.map(({ id, eventname, date, description, largeimageurl, venue}) => <SearchEvent 
                            key={id} 
                            id={id}
                            name={eventname} 
                            date={date} 
                            desc={description} 
                            img={largeimageurl} 
                            venue={venue.name} />)
                    }
                    </div>
                </div>
            </div>
        )
    }
}

export default Search;