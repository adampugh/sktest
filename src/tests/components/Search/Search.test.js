import React from 'react';
import '../../setupTests';
import { shallow } from 'enzyme';

import Search from '../../../components/Search/Search';

describe("<Search />", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(<Search />);
        expect(wrapper).toMatchSnapshot();
    });
});