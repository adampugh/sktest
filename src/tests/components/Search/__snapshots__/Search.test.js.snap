// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search__hero"
        >
          <div
            className="container"
          >
            <img
              alt="logo"
              className="search__hero__logo"
              src="logo.png"
            />
            <input
              className="search__hero__input"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="Search for events"
              value=""
            />
          </div>
        </div>,
        <div
          className="search__events"
        >
          <div
            className="search__events__grid container"
          />
        </div>,
      ],
      "className": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <img
              alt="logo"
              className="search__hero__logo"
              src="logo.png"
            />
            <input
              className="search__hero__input"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="Search for events"
              value=""
            />
          </div>,
          "className": "search__hero",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="search__hero__logo"
                src="logo.png"
              />,
              <input
                className="search__hero__input"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Search for events"
                value=""
              />,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "search__hero__logo",
                "src": "logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search__hero__input",
                "onChange": [Function],
                "onKeyPress": [Function],
                "placeholder": "Search for events",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="search__events__grid container"
          />,
          "className": "search__events",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "search__events__grid container",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search__hero"
          >
            <div
              className="container"
            >
              <img
                alt="logo"
                className="search__hero__logo"
                src="logo.png"
              />
              <input
                className="search__hero__input"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Search for events"
                value=""
              />
            </div>
          </div>,
          <div
            className="search__events"
          >
            <div
              className="search__events__grid container"
            />
          </div>,
        ],
        "className": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <img
                alt="logo"
                className="search__hero__logo"
                src="logo.png"
              />
              <input
                className="search__hero__input"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Search for events"
                value=""
              />
            </div>,
            "className": "search__hero",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className="search__hero__logo"
                  src="logo.png"
                />,
                <input
                  className="search__hero__input"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Search for events"
                  value=""
                />,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": "search__hero__logo",
                  "src": "logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "search__hero__input",
                  "onChange": [Function],
                  "onKeyPress": [Function],
                  "placeholder": "Search for events",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="search__events__grid container"
            />,
            "className": "search__events",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "search__events__grid container",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
